src/main.cpp

```diff
@@ -2398,11 +2398,16 @@ bool static DisconnectTip(CValidationState &state) {
     mempool.check(pcoinsTip);
     // Update chainActive and related variables.
     UpdateTip(pindexDelete->pprev);
+    // Get the current commitment tree
+    ZCIncrementalMerkleTree newTree;
+    assert(pcoinsTip->GetAnchorAt(pcoinsTip->GetBestAnchor(), newTree));
```

Comment:
Do we really need to send the tree here? I don't thin ChainTip/DecrementWitnesses does anything with it, does it?